src/main/java/com/opinta/service/PDFGeneratorServiceImpl.java

```diff
@@ -64,76 +138,136 @@ public PDFGeneratorServiceImpl(ShipmentService shipmentService) {
+            populateField(acroForm, field, "weight", String.valueOf(shipment.getWeight()));
+            populateField(acroForm, field, "declaredPrice", String.valueOf(shipment.getDeclaredPrice()));
+            populateField(acroForm, field, "postPay", String.valueOf(shipment.getPostPay()));
+            populateField(acroForm, field, "price", String.valueOf(shipment.getPrice()));
 
-                field = (PDTextField) acroForm.getField("sendingCost");
-                field.setValue(String.valueOf(shipment.getPrice()));
+            //TODO: Not interactive fields yet!
+            populateField(acroForm, field, "sendingCost", String.valueOf(shipment.getPrice()));
+            populateField(acroForm, field, "additionalCosts", "0");
 
-                field = (PDTextField) acroForm.getField("postPrice");
-                field.setValue(String.valueOf(shipment.getPostPay()));
+            //Creating content stream for the page to allow data appending
+            PDPage page = template.getPage(0);
+            PDPageContentStream contentStream =
+                    new PDPageContentStream(template, page, PDPageContentStream.AppendMode.APPEND, true);
 
-                field = (PDTextField) acroForm.getField("totalCost");
-                field.setValue(String.valueOf(shipment.getPostPay()));
-            }
+            //Constructing 13 digits of the barcode
+            String barcode = shipment.getSender().getCounterparty().getPostcodePool().getPostcode() +
+                    shipment.getBarcode().getInnerNumber();
+
+            //Generating first barcode
+            bitMatrix = new Code128Writer().encode(barcode, BarcodeFormat.CODE_128, 170, 32, null);
+            BufferedImage buffImg = MatrixToImageWriter.toBufferedImage(bitMatrix);
+            PDImageXObject ximage = JPEGFactory.createFromImage(template, buffImg);
+            contentStream.drawImage(ximage, 242, 790);
+
+            //Generating second barcode
+            bitMatrix = new Code128Writer().encode(barcode, BarcodeFormat.CODE_128, 170, 32, null);
+            buffImg = MatrixToImageWriter.toBufferedImage(bitMatrix);
+            ximage = JPEGFactory.createFromImage(template, buffImg);
+            contentStream.drawImage(ximage, 242, 377);
+
+            contentStream.close();
+
+            //Generating barcode digits
+            field = (PDTextField) acroForm.getField("barcodeText");
+            field.setDefaultAppearance(String.format("/%s 14 Tf 0 g", fontName));
+            field.setValue(barcode);
+
+            //Removing acrofiels
+            acroForm.flatten();
+
+            //Saving PDF to byte array
             ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
             template.save(outputStream);
             data = outputStream.toByteArray();
+            outputStream.close();
         } catch (IOException e) {
             log.error("Error while parsing PDF template: " + e.getMessage());
```

Comment:
Plz, use something like this: log.error("Some message with parameter {}", myParam)